library/Graphite/Web/Widget/GraphImage.php

```diff
@@ -126,7 +126,7 @@ public function render()
                         // Try to render a higher time range, but give up
                         // once our default (1h) has been reached (non successfully).
                         if ($diff < 3600) {
-                            $url->setParam('from', $until - $diff * 2);
+                            $url->setParam('from', (string) ($until - $diff * 2));
```

Comment:
Please use `sprintf('%F', ...)` here. The string cast is locale dependent prior PHP 8.0.